incubator/binding-risingwave/src/main/java/io/aklivity/zilla/runtime/binding/risingwave/internal/macro/RisingwaveCreateStreamMacro.java

```diff
@@ -124,6 +143,8 @@ public RisingwaveMacroState onError(
             PgsqlFlushExFW flushEx)
         {
             handler.doFlushProxy(traceId, authorization, flushEx);
+            onReadyHandler = terminateState;
+
             return this;
```

Comment:
Shouldn't this return an error state instead, which calls `handler.doReady(...)` for `onReady(...)`?
Then I think we don't need to introduce a separate ready handler interface as the transition is captured by moving to a different state.